core/capabilities.go

```diff
@@ -20,16 +21,21 @@ type ModelConstraint struct {
 type Capability int
 type CapabilityString []uint64
 type Constraints struct {
+	minVersion string
+}
+type PerCapabilityConstraints struct {
 	// Models contains a *ModelConstraint for each supported model ID
 	Models ModelConstraints
 }
-type CapabilityConstraints map[Capability]*Constraints
+type CapabilityConstraints map[Capability]*PerCapabilityConstraints
```

Comment:
@victorges feel free to suggest naming improvements if needed 👍🏻.

Reply:
Let's keep as is and possibly refactor later/on a separate PR merged here.